pkg/registrymirror/registrymirrorsstore.go

```diff
@@ -0,0 +1,325 @@
+package registrymirror
+
+import (
+	"bytes"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/BurntSushi/toml"
+	"github.com/containers/image/v5/pkg/sysregistriesv2"
+	ciTypes "github.com/containers/image/v5/types"
+	"github.com/docker/distribution/reference"
+	configV1 "github.com/openshift/api/config/v1"
+	operatorV1Alpha1 "github.com/openshift/api/operator/v1alpha1"
+	"github.com/openshift/runtime-utils/pkg/registries"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/maputil"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stackrox/rox/pkg/utils"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	// default path to store registries config.
+	defaultRegistriesPath = "/var/cache/stackrox/mirrors/registries.conf"
+
+	// default delay before writing the updated registries config.
+	defaultDelay = time.Millisecond * 300
```

Comment:
nit: usually we do `300 * time.Millisecond`

Also, where did this come from? just curious

Reply:
> Also, where did this come from? just curious

I took a look at the time it takes from when the SYNC events are received by the handler to when syncing is completed. On average was taking about 2 seconds prior to the initial sync being completed - so figured that was enough time.  However looking at the SYNC events they are received within microseconds of each other so I'll drop this down further.

Reply:
```
kubernetes/listener/resources: 2023/08/09 18:39:52.355998 registrymirrorsets.go:52: Debug: Upserted ImageContentSourcePolicy into mirror store: "icspb" (399f39a0-4994-4cc5-80a0-73cdd0394832)
kubernetes/listener/resources: 2023/08/09 18:39:52.356175 registrymirrorsets.go:52: Debug: Upserted ImageContentSourcePolicy into mirror store: "icspc" (f309096f-c52a-4d25-8be0-43345b10494b)
kubernetes/listener/resources: 2023/08/09 18:39:52.356244 registrymirrorsets.go:52: Debug: Upserted ImageContentSourcePolicy into mirror store: "icspd" (a70ecc3a-33cf-42bd-b106-8af6d54e4eb7)
kubernetes/listener/resources: 2023/08/09 18:39:52.356362 registrymirrorsets.go:52: Debug: Upserted ImageContentSourcePolicy into mirror store: "icspe" (d540e3f0-88d7-4b0c-b8f0-59a1b0729465)
kubernetes/listener/resources: 2023/08/09 18:39:52.356873 registrymirrorsets.go:52: Debug: Upserted ImageContentSourcePolicy into mirror store: "icspa" (a2d87863-3869-4125-a347-b004ff346dad)
pkg/registrymirror: 2023/08/09 18:39:52.407431 store.go:192: Debug: Successfully updated the registries config at "/var/cache/stackrox/mirrors/registries.conf" with 5 icspRules, 0 idmsRules, 0 itmsRules
```
On initial sync they updates are micro seconds apart, if applying the changes via `kubectl apply/delete -f ...` and the file contains multiple CRs - the time between updates varies (if multiple resources in same file believe each resources is separate API request):
```
kubernetes/listener/resources: 2023/08/09 18:43:06.769103 registrymirrorsets.go:44: Debug: Deleted ImageContentSourcePolicy from mirror store: "icspa" (a2d87863-3869-4125-a347-b004ff346dad)
kubernetes/listener/resources: 2023/08/09 18:43:06.818298 registrymirrorsets.go:44: Debug: Deleted ImageContentSourcePolicy from mirror store: "icspb" (399f39a0-4994-4cc5-80a0-73cdd0394832)
kubernetes/listener/resources: 2023/08/09 18:43:06.869845 registrymirrorsets.go:44: Debug: Deleted ImageContentSourcePolicy from mirror store: "icspc" (f309096f-c52a-4d25-8be0-43345b10494b)
pkg/registrymirror: 2023/08/09 18:43:06.869907 store.go:192: Debug: Successfully updated the registries config at "/var/cache/stackrox/mirrors/registries.conf" with 3 icspRules, 0 idmsRules, 0 itmsRules
kubernetes/listener/resources: 2023/08/09 18:43:06.919646 registrymirrorsets.go:44: Debug: Deleted ImageContentSourcePolicy from mirror store: "icspd" (a70ecc3a-33cf-42bd-b106-8af6d54e4eb7)
pkg/registrymirror: 2023/08/09 18:43:06.970872 store.go:192: Debug: Successfully updated the registries config at "/var/cache/stackrox/mirrors/registries.conf" with 1 icspRules, 0 idmsRules, 0 itmsRules
kubernetes/listener/resources: 2023/08/09 18:43:06.974610 registrymirrorsets.go:44: Debug: Deleted ImageContentSourcePolicy from mirror store: "icspe" (d540e3f0-88d7-4b0c-b8f0-59a1b0729465)
pkg/registrymirror: 2023/08/09 18:43:07.028288 store.go:192: Debug: Successfully updated the registries config at "/var/cache/stackrox/mirrors/registries.conf" with 0 icspRules, 0 idmsRules, 0 itmsRules
```